### PR TITLE
added to generate-thumbnail example to upload Signed URLs to the database

### PR DESCRIPTION
**Pull request by jenperson**

### CODE REVIEW

**Comment by jenperson** [comment]
I signed it!

**Comment by jenperson** [comment]
I signed it!

File: generate-thumbnail/functions/index.js [hunk 1]
```diff
@@ -62,21 +70,40 @@ exports.generateThumbnail = functions.storage.object().onChange(event => {
...
+  }).then(results => {
+    console.log('Got Signed URL');
+    const thumbResult = results[0];
+    const originalResult = results[1];
+    const thumbFileUrl = thumbResult[0];
+    const fileUrl = originalResult[0];
+    // Add the URLs to the Database
+    return ref.child("images").push({path: fileUrl, thumbnail: thumbFileUrl});
+  }).catch(reason => {
+    console.log(reason);
```
Comment (nicolasgarnier) [inline generate-thumbnail/functions/index.js]: Could you change that to `console.error(reason)` ?

File: generate-thumbnail/functions/index.js [hunk 2]
```diff
@@ -62,21 +70,40 @@ exports.generateThumbnail = functions.storage.object().onChange(event => {
...
+      file.getSignedUrl(config),
+    ]);
+  }).then(results => {
+    console.log('Got Signed URL');
+    const thumbResult = results[0];
+    const originalResult = results[1];
+    const thumbFileUrl = thumbResult[0];
+    const fileUrl = originalResult[0];
+    // Add the URLs to the Database
+    return ref.child("images").push({path: fileUrl, thumbnail: thumbFileUrl});
```
Comment (nicolasgarnier) [inline generate-thumbnail/functions/index.js]: Switch to single quotes for Strings.

File: generate-thumbnail/functions/index.js [hunk 3]
```diff
@@ -26,20 +28,26 @@ const THUMB_MAX_HEIGHT = 200;
 const THUMB_MAX_WIDTH = 200;
 // Thumbnail prefix added to file names.
 const THUMB_PREFIX = 'thumb_';
+admin.initializeApp(functions.config().firebase);
+const ref = admin.database().ref();
 
 /**
  * When an image is uploaded in the Storage bucket We generate a thumbnail automatically using
  * ImageMagick.
```
Comment (nicolasgarnier) [inline generate-thumbnail/functions/index.js]: Could you add a sentence about the signedURL in the function comment here? something like: "After the thumbnail has been generated and uploaded to Cloud Storage we write the public URL to the Firebase Realtime Database."

Comment (jenperson) [inline generate-thumbnail/functions/index.js]: I've made changes based upon your suggestions. Thank you!

**Comment by nicolasgarnier** [comment]
Looks good thanks @jenperson!

**Comment by NainalChauhan** [comment]
The generated SignedUrl expires after 7 days. It works perfect for some days and then throws SignatureDoesNotMatch error. Is there any solution ?

**Pull request merged**